### PR TITLE
Add the HasLen checker.

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

File: checker.go [hunk 1]
```diff
@@ -305,6 +305,51 @@ func (c *isNilChecker) Negate(got interface{}, args []interface{}) error {
...
+
+type hasLenChecker struct {
+	numArgs
+}
+
+// Check implements Checker.Check by checking that len(got) == args[0].
+func (c *hasLenChecker) Check(got interface{}, args []interface{}) (err error) {
+	want, ok := args[0].(int)
+	if !ok {
+		return BadCheckf("expected a numeric length to compare the value to, got %T instead", args[0])
```
Comment (rogpeppe) [inline checker.go]: "numeric length" isn't very specific.


How about: "expected length is of type %T, not int" ?

File: checker.go [hunk 2]
```diff
@@ -305,6 +305,51 @@ func (c *isNilChecker) Negate(got interface{}, args []interface{}) error {
...
+	}
+	v := reflect.ValueOf(got)
+	switch v.Kind() {
+	case reflect.Array, reflect.Chan, reflect.Map, reflect.Slice, reflect.String:
+	default:
+		return BadCheckf("expected a type with a lenght, got %T instead", got)
```
Comment (rogpeppe) [inline checker.go]: s/lenght/length/

File: checker.go [hunk 3]
```diff
@@ -305,6 +305,51 @@ func (c *isNilChecker) Negate(got interface{}, args []interface{}) error {
...
+	}
+	if length := v.Len(); length != want {
+		return fmt.Errorf("the provided value has not the expected length of %d:\n(value)\n\t%#v\n(-got length +want lenght)\n\t-: %d\n\t+: %d", want, got, length, want)
```
Comment (rogpeppe) [inline checker.go]: s/lenght/length/

I wonder whether it might be good to print the contents of the slice too.

File: checker.go (hunk 2 above)
Comment (frankban) [reply checker.go]: Done.

File: checker.go (hunk 1 above)
Comment (frankban) [reply checker.go]: Done.

File: checker.go (hunk 3 above)
Comment (frankban) [reply checker.go]: Already there.

**Comment by frankban** [comment]
Thanks for the review!

**Pull request merged**